rest/config.go

```diff
@@ -1603,8 +1661,17 @@ func (sc *ServerContext) FetchConfigs(ctx context.Context, isInitialStartup bool
 			// inherit properties the bootstrap config
 			cnf.CACertPath = sc.Config.Bootstrap.CACertPath
 
+			// We need to check for corruption in the database config (CC. CBG-3292). If the fetched config doesn't match the
+			// bucket name we got the config from we need to maker this db context as corrupt. Then remove the context and
+			// in memory representation on the server context.
 			bucketCopy := bucket
-			cnf.Bucket = &bucketCopy
+			if bucket != *cnf.Bucket {
+				sc.handleInvalidDatabaseConfig(ctx, bucket, *cnf)
+				continue
+			} else {
+				// no corruption detected carry on as usual
+				cnf.Bucket = &bucketCopy
+			}
```

Comment:
```suggestion
			cnf.Bucket = &bucketCopy
```

we could drop the else statement here for readability (nit)